clippy_lints/src/transmute.rs

```diff
@@ -226,263 +253,293 @@ declare_lint_pass!(Transmute => [
     TRANSMUTE_BYTES_TO_STR,
     TRANSMUTE_INT_TO_BOOL,
     TRANSMUTE_INT_TO_FLOAT,
+    UNSOUND_COLLECTION_TRANSMUTE,
 ]);
 
+// used to check for UNSOUND_COLLECTION_TRANSMUTE
+static COLLECTIONS: &[&[&str]] = &[
```

Comment:
Should we also include `LinkedList` here?

Reply:
I'm not sure. It might be admissible **IF** the alignment&size of the target type is lower or equal to that of the source type, the list isn't modified and either leaked or transmuted back to the original type afterwards. But this hasn't yet been specified by the unsafe WG, so I wouldn't bet on it. In any event, I'm erring on the side of least false positives here.